Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,438 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+  mkNatIso (Iso.refl _) (Iso.refl _)
+
+end NatTrans
+
+/-- A `reflexivePair` composed with a functor is isomorphic to the `reflexivePair` obtained by
+applying the functor at each map. -/
+@[simps!]
+def compRightIso {D : Type u₂} [Category.{v₂} D] {A B : C}
+    (f g : A ⟶ B) (s : B ⟶ A) (sl : s ≫ f = 𝟙 B) (sr : s ≫ g = 𝟙 B) (F : C ⥤ D) :
+    (reflexivePair f g s sl sr) ⋙ F ≅ reflexivePair (F.map f) (F.map g) (F.map s)
+      (by simp only [← Functor.map_comp, sl, Functor.map_id])
+      (by simp only [← Functor.map_comp, sr, Functor.map_id]) :=
+  mkNatIso (Iso.refl _) (Iso.refl _)
+
+lemma whiskerRightMkNatTrans {F G: WalkingReflexivePair ⥤ C}
+    (e₀ : F.obj zero ⟶ G.obj zero) (e₁ : F.obj one ⟶ G.obj one)
+    {h₁ : F.map left ≫ e₀ = e₁ ≫ G.map left}
+    {h₂ : F.map right ≫ e₀ = e₁ ≫ G.map right}
+    {h₃ : F.map reflexion ≫ e₁ = e₀ ≫ G.map reflexion}
+    {D : Type u₂} [Category.{v₂} D] (H : C ⥤ D) :
+    whiskerRight (mkNatTrans e₀ e₁ : F ⟶ G) H =
+      mkNatTrans (H.map e₀) (H.map e₁)
+          (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₁])
+          (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₂])
+          (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₃]) := by
+  ext x; cases x <;> simp
+
+end
+
+/-- Any functor out of the WalkingReflexivePair yields a reflexive pair -/
+instance to_isReflexivePair {F : WalkingReflexivePair ⥤ C} :
+    IsReflexivePair (F.map .left) (F.map .right) :=
+  ⟨F.map .reflexion, map_reflexion_comp_map_left F, map_reflexion_comp_map_right F⟩
+
+end reflexivePair
+
+/-- A `ReflexiveCofork` is a cocone over a `WalkingReflexivePair`-shaped diagram. -/
+abbrev ReflexiveCofork (F : WalkingReflexivePair ⥤ C) := Cocone F
+
+namespace ReflexiveCofork
+
+open WalkingReflexivePair WalkingReflexivePair.Hom
+
+variable {F : WalkingReflexivePair ⥤ C}
+
+/-- The tail morphism of a reflexive cofork. -/
+abbrev π (G : ReflexiveCofork F) : F.obj zero ⟶ G.pt := G.ι.app zero
+
+/-- Constructor for `ReflexiveCofork` -/
+@[simps pt]
+def mk {X : C} (π : F.obj zero ⟶ X) (h : F.map left ≫ π = F.map right ≫ π) :
+    ReflexiveCofork F where
+  pt := X
+  ι := reflexivePair.mkNatTrans π (F.map left ≫ π)
+
+@[simp]
+lemma mk_π {X : C} (π : F.obj zero ⟶ X) (h : F.map left ≫ π = F.map right ≫ π) :
+    (mk π h).π = π := rfl
+
+lemma condition (G : ReflexiveCofork F) : F.map left ≫ G.π = F.map right ≫ G.π := by
+  erw [Cocone.w G left, Cocone.w G right]
+
+@[simp]
+lemma app_one_eq_π (G : ReflexiveCofork F) : G.ι.app zero = G.π := rfl
+
+/-- The underlying `Cofork` of a `ReflexiveCofork`. -/
+def toCofork (G : ReflexiveCofork F) : Cofork (F.map left) (F.map right) :=
+  Cocones.precompose (diagramIsoParallelPair (_ ⋙ F)).symm.hom |>.obj <|
+    G.whisker WalkingParallelPair.inclusionWalkingReflexivePair
+
+@[simp]
+lemma toCofork.π (G : ReflexiveCofork F) : Cofork.π (G.toCofork) = G.π := by
+  dsimp only [toCofork, Cofork.π]
+  aesop_cat
```

Comment:
I would favour a more down to earth definition here:
```suggestion
abbrev toCofork (G : ReflexiveCofork F) : Cofork (F.map left) (F.map right) :=
  Cofork.ofπ G.π (by simp)
```

Then, `inclusionWalkingReflexivePairEquiv_obj` below become an iso (use `Cofork.ext`) instead of an equality, and the definition of `ReflectiveCofork.isColimitEquiv` would have to be changed, using the composition with `IsColimit.equivIsoColimit (inclusionWalkingReflexivePairEquivObjIso G).symm)`.